betty/service/__init__.py

```diff
@@ -0,0 +1,58 @@
+"""
+Provide a service container.
+"""
+from __future__ import annotations
+
+from collections.abc import Callable
+from types import TracebackType
+from typing import TypeAlias, Any, Self, overload, AsyncContextManager
+
+ServiceId: TypeAlias = str
+Service: TypeAlias = Any
+ServiceFactory: TypeAlias = Callable[['ServiceContainer'], AsyncContextManager[Service]]
+
+
+class _CyclicDependency(RuntimeError):
+    pass
+
+
+class _ServiceNotFound(RuntimeError):
+    pass
+
+
+class _ServiceContainerNotStarted(RuntimeError):
+    pass
+
+
+class ServiceContainer:
```

Comment:
Make service containers take and expose a context, over which type containers are generic.

Reply:
This should internally be a weakref, so a context can own the container.